shinkai-bin/shinkai-node/src/managers/tool_router.rs

```diff
@@ -631,11 +632,31 @@ impl ToolRouter {
         unique_files.extend(additional_files.into_iter());
         let all_files: Vec<_> = unique_files.into_iter().collect();
 
+        let agent_id = if let ProviderOrAgent::Agent(agent) = context.agent() {
+            Some(agent.clone().agent_id)
+        } else {
+            None
+        };
+
+        // If agent_id is provided, get the agent's tool config overrides and merge with extra_config
+        let function_config = shinkai_tool.get_config_from_env();
+        let mut function_config_vec: Vec<ToolConfig> = function_config.into_iter().collect();
+        let agent = context.agent().clone();
+        match agent {
```

Comment:
we dont need to update it but we can do an assignment let something = match agent { ...